SDL2_OpenGL_BoilerPlate/BoilerPlate/Entity.hpp

```diff
@@ -8,16 +11,29 @@ class Entity {
 protected:
 	Vector2* position;
 	std::vector<Vector2> entityVertexContainer;
+	Vector2* entityVelocity;
+	float mass;
+	float speed;
+	float radius;
+	bool debuggerState;
 
 
 public:
 	//Constructor
 	Entity();
 
 	//Functions
-	void Update();
+	void Update(int screenWidth, int screenHeight, float deltaTime);
```

Comment:
I'd recommend trying to find a way of removing references to the width and height from the update overloads. Think of making entities have a parent which holds this info.